src/nvim/option.c

```diff
@@ -5913,15 +5919,16 @@ option_value2string (
   varp = get_varp_scope(opp, opt_flags);
 
   if (opp->flags & P_NUM) {
-    long wc = 0;
+    NumOpt wc = 0;
 
-    if (wc_use_keyname(varp, &wc))
+    if (wc_use_keyname(varp, &wc)) {
       STRCPY(NameBuff, get_special_key_name((int)wc, 0));
-    else if (wc != 0)
+    } else if (wc != 0) {
       STRCPY(NameBuff, transchar((int)wc));
-    else
-      sprintf((char *)NameBuff, "%" PRId64, (int64_t)*(long *)varp);
-  } else { /* P_STRING */
+    } else {
+      sprintf((char *)NameBuff, "%" PRId64, (int64_t)*(NumOpt *)varp); // NOLINT
```

Comment:
What did you do and what do you mean by “not worked”? If you create a define you should use it (and remove the cast to `int64_t`), if you use intmax_t you should change PRId64 to PRIdMAX. `intmax_t` is valid to be `long` on e.g. 64-bit linux.